components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -520,20 +532,22 @@ private String handleOAuthAuthorizationRequest(String clientId, HttpServletReque
 
         String sessionDataKey = UUIDGenerator.generateUUID();
         CacheKey cacheKey = new SessionDataCacheKey(sessionDataKey);
-        sessionDataCacheEntry = new SessionDataCacheEntry();
-        sessionDataCacheEntry.setoAuth2Parameters(params);
-        sessionDataCacheEntry.setQueryString(req.getQueryString());
+        SessionDataCacheEntry sessionDataCacheEntryNew = new SessionDataCacheEntry();
+        sessionDataCacheEntryNew.setoAuth2Parameters(params);
+        sessionDataCacheEntryNew.setQueryString(req.getQueryString());
 
         if (req.getParameterMap() != null) {
-            sessionDataCacheEntry.setParamMap(new ConcurrentHashMap<String, String[]>(req.getParameterMap()));
+            sessionDataCacheEntryNew.setParamMap(new ConcurrentHashMap<String, String[]>(req.getParameterMap()));
         }
-        SessionDataCache.getInstance().addToCache(cacheKey, sessionDataCacheEntry);
+        SessionDataCache.getInstance().addToCache(cacheKey, sessionDataCacheEntryNew);
 
         try {
             return EndpointUtil.getLoginPageURL(clientId, sessionDataKey, forceAuthenticate,
                     checkAuthentication, oauthRequest.getScopes(), req.getParameterMap());
         } catch (UnsupportedEncodingException e) {
-            log.debug(e.getMessage(), e);
+            if (log.isDebugEnabled()) {
+                log.debug("Error while retrieving the login page url.", e);
```

Comment:
Any reason why we have this defined as a debug log ? it might be better to explain it in a comment.